### ISSUE TITLE
svg2png generates corrupt image

### ISSUE THREAD

**Issue by brendabell** [Kozea/CairoSVG#299]
I'm trying to convert a working app from Python 2.7 tot 3.7. The app was working fine with the older Python.

I know Python 3 handles strings differently so I've verified that my svg string is a python str (not bytes and not unicode). I'm using cairosvg 2.4.2.

I've tried the following:

`cairosvg.svg2png(bytestring=result.encode('utf-8'))`
`cairosvg.svg2png(bytestring=bytes(result, 'utf-8'))`
`cairosvg.svg2png(bytestring=result)`

The result is a png file that can't be opened by the Preview app (I'm on Mac OS X 10.14).

What am I doing wrong?

**Comment by liZe** [comment]
Hi!

If your image is corrupted, it’s probably not because of the SVG input (the `bytestring` parameter), but because of the output. `svg2png` returns a bytestring, and maybe there’s a problem with what you do with this bytestring.

**Comment by brendabell** [comment]
> Hi!
> 
> If your image is corrupted, it’s probably not because of the SVG input (the `bytestring` parameter), but because of the output. `svg2png` returns a bytestring, and maybe there’s a problem with what you do with this bytestring.

This was an awesome clue, thanks. It made me think about the changes I had to do to support python 3 -- I got bit by the string vs bytes change. My other response generation functions are returning strings so I was converting the strings to bytes as required by `BaseHTTPRequestHandler.wfile.write` immediately before writing the content.

In short, that conversion was poorly placed and I was doing an unnecessary conversion on the PNG content. All fixed and working perfectly.
